lib/experimental/fonts/font-library/class-wp-rest-font-faces-controller.php

```diff
@@ -0,0 +1,762 @@
+		if ( ! function_exists( 'wp_handle_upload' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		$srcs           = is_string( $settings['src'] ) ? array( $settings['src'] ) : $settings['src'];
+		$processed_srcs = array();
+		$font_file_meta = array();
+
+		foreach ( $srcs as $src ) {
+			// If src not a file reference, use it as is.
+			if ( ! isset( $file_params[ $src ] ) ) {
+				$processed_srcs[] = $src;
+				continue;
+			}
+
+			$file      = $file_params[ $src ];
+			$font_file = $this->handle_font_file_upload( $file );
+			if ( is_wp_error( $font_file ) ) {
+				return $font_file;
+			}
+
+			$processed_srcs[] = $font_file['url'];
+			$font_file_meta[] = $this->relative_fonts_path( $font_file['file'] );
+		}
+
+		// Store the updated settings for prepare_item_for_database to use.
+		$settings['src'] = count( $processed_srcs ) === 1 ? $processed_srcs[0] : $processed_srcs;
+		$request->set_param( 'font_face_settings', $settings );
+
+		// Ensure that $settings data is slashed, so values with quotes are escaped.
+		// WP_REST_Posts_Controller::create_item uses wp_slash() on the post_content.
+		$font_face_post = parent::create_item( $request );
+
+		if ( is_wp_error( $font_face_post ) ) {
+			return $font_face_post;
+		}
+
+		$font_face_id = $font_face_post->data['id'];
+
+		foreach ( $font_file_meta as $font_file_path ) {
+			add_post_meta( $font_face_id, '_wp_font_face_file', $font_file_path );
+		}
+
+		return $font_face_post;
+	}
+
+	/**
+	 * Deletes a single font face.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
+
+		// We don't support trashing for revisions.
+		if ( ! $force ) {
+			return new WP_Error(
+				'rest_trash_not_supported',
+				/* translators: %s: force=true */
+				sprintf( __( "Font faces do not support trashing. Set '%s' to delete.", 'gutenberg' ), 'force=true' ),
+				array( 'status' => 501 )
+			);
+		}
+
+		return parent::delete_item( $request );
+	}
+
+	/**
+	 * Prepares a single font face output for response.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Post         $item    Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable -- required by parent class
+		$data = array();
+
+		$data['id']                 = $item->ID;
+		$data['theme_json_version'] = 2;
+		$data['parent']             = $item->post_parent;
+		$data['font_face_settings'] = $this->get_settings_from_post( $item );
+
+		$response = rest_ensure_response( $data );
+		$links    = $this->prepare_links( $item );
+		$response->add_links( $links );
```

Comment:
```suggestion
                 if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
		    $links    = $this->prepare_links( $item );
		    $response->add_links( $links );
		}
```

As noted above, only prepare and add links if requested.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57688/commits/32689e176c2d2df55803a1fa06903ac148e8df7d